src/client/Microsoft.Identity.Client/TelemetryCore/Internal/Events/ApiEvent.cs

```diff
@@ -63,164 +35,102 @@ public enum ApiIds
-                                        ? HashPersonalIdentifier(_cryptographyManager, value)
-                                        : null;
-        }
+        public string TenantId { get; set; }
 
-        public string AccountId
-        {
-            set =>
-                this[UserIdKey] = value != null && _logger.PiiLoggingEnabled
-                                      ? HashPersonalIdentifier(_cryptographyManager, value)
-                                      : null;
-        }
+        public string AccountId { get; set; }
 
-        public bool WasSuccessful
-        {
-#pragma warning disable CA1305 // .net standard does not have an overload for ToString() with Culture
-            set { this[WasSuccessfulKey] = value.ToString().ToLowerInvariant(); }
-            get { return this[WasSuccessfulKey] == true.ToString().ToLowerInvariant(); }
-#pragma warning restore CA1305 // Specify IFormatProvider
+        public bool WasSuccessful { get; set; }
 
-        }
+        public bool IsConfidentialClient { get; set; }
 
-        public bool IsConfidentialClient
-        {
-#pragma warning disable CA1305 // Specify IFormatProvider
-            set { this[IsConfidentialClientKey] = value.ToString().ToLowerInvariant(); }
-#pragma warning restore CA1305 // Specify IFormatProvider
-        }
+        public bool IsAccessTokenCacheHit { get; set; }
 
-        public bool IsAccessTokenCacheHit
-        {
-#pragma warning disable CA1305 // Specify IFormatProvider
-            get
-            {  return this.ContainsKey(IsAccessTokenCacheHitKey) ?
-                    (this[IsAccessTokenCacheHitKey] == true.ToString().ToLowerInvariant()) : 
-                    false; }
-            set { this[IsAccessTokenCacheHitKey] = value.ToString().ToLowerInvariant(); }
-#pragma warning restore CA1305 // Specify IFormatProvider
-        }
+        public string ApiErrorCode { get; set; }
 
-        public string ApiErrorCode
-        {
-            get => this.ContainsKey(ApiErrorCodeKey) ? this[ApiErrorCodeKey] : null;
-            set => this[ApiErrorCodeKey] = value;
-        }
+        public string LoginHint { get; set; }
 
-        public string LoginHint
-        {
-            set =>
-                this[LoginHintKey] = value != null && _logger.PiiLoggingEnabled
-                                         ? HashPersonalIdentifier(_cryptographyManager, value)
-                                         : null;
-        }
+        #region Region
+        public string RegionUsed { get; set; }
 
-#region Region
-        public string RegionUsed
+        private int? _regionAutodetectionSource;
+        public int RegionAutodetectionSource
         {
-            get => this.ContainsKey(RegionUsedKey) ? this[RegionUsedKey] : null;
-            set => this[RegionUsedKey] = value;
+            get { return _regionAutodetectionSource.HasValue ? _regionAutodetectionSource.Value : 0; }
```

Comment:
nit: smaller code _regionAutodetectionSource ?? 0;

Reply:
Fixed in PR #3058.